unskript-ctl/templates/template_script.j2

```diff
@@ -26,18 +29,39 @@ def _run_function(fn):
     else:
         try:
             fn = globals().get(fn)
+            if _logger:
+                # Since lego and lego_printer are nested within the check function
+                # we can use co_consts to extract the lego_printer and strip off the 
+                # _printer to get the lego function name
+                chk_name = ''
+                for const in fn.__code__.co_consts:
+                    if isinstance(const, type(fn.__code__)):
+                        if const.co_name.endswith('printer'):
+                            chk_name = const.co_name.replace('_printer','')
+                _logger.debug(f"Starting to execute check {fn} {chk_name}")
```

Comment:
Also, in the test logs, I didn't see output with multiple checks being run.
It is not clear to me that when we run with multiple checks, why this code will pick up the correct check name (since there will be multiple functions that end with `_printer`)

Reply:
In fact the test outputs do show that the same name is being printed four times instead of four names.